Server/Application/SpringBoard.m

```diff
@@ -187,22 +187,29 @@ - (SpringBoardAlertHandlerResult)handleAlert {
     }
 
     XCUIElement *button = nil;
-    NSString *mark = springBoardAlert.defaultDismissButtonMark;
+    NSArray *marks = springBoardAlert.defaultDismissButtonMarks;
 
     // Alert is now gone? It can happen...
     if (!alert.exists) {
         return SpringBoardAlertHandlerNoAlert;
     }
+        
+    for (NSString *mark in marks) {
+        button = alert.buttons[mark];
 
-    button = alert.buttons[mark];
-    // Resolve before asking if the button exists.
-    if ([button respondsToSelector:@selector(resolve)]) {
-        [button resolve];
-    } else {
-        NSError *error = nil;
-        if (![button resolveOrRaiseTestFailure:NO error:&error]) {
-            DDLogWarn(@"Encountered an error resolving element '%@':\n%@",
-                      button, [error localizedDescription]);
+        // Resolve before asking if the button exists.
+        if ([button respondsToSelector:@selector(resolve)]) {
+            [button resolve];
+        } else {
+            NSError *error = nil;
+            if (![button resolveOrRaiseTestFailure:NO error:&error]) {
```

Comment:
Let's simplify this and remove the error handling since we are expecting that `NO` means no error will be raised.

Reply:
Do you mean that we need to keep only `[button resolveOrRaiseTestFailure]`?

Reply:
Search of dismiss button was moved in a separate function and error handling were removed there